### ISSUE TITLE
creating snap files from another processing pipeline

### ISSUE THREAD

**Issue by dawe** [r3fang/SnapTools#16]
I would like to process my scATAC data with snaptools and snapatac but, for several reasons, the BAM files I have are in a format which is different from the one required by snaptools (i.e. the cell barcode is not in the read name and it's a RG tag instead). In my BAM files, reads are already deduplicated at cell level (that is, the BAM flag is properly set). I already have counts over binned genome (at 5kb) in a scipy sparse matrix. What would be the best way to convert my matrices into snap format?

**Comment by r3fang** [comment]
Hi there,

There are multiple entry point that you can generate the snap file. The easiest way is to convert the bam file to the format that required by sanptools. Here is the example:
https://github.com/r3fang/SnapATAC/wiki/FAQs#cellranger_output <https://github.com/r3fang/SnapATAC/wiki/FAQs#cellranger_output>

Best
-Rongxin

**Comment by dawe** [comment]
Yes, I did that as it sounds the only way to go. Still I’d rather not use read-sorted bam (it wastes disk space and time in sorting, IMHO), my BAM files already have barcode names in their header and read groups, so the size of the resulting matrix is known from the start and reads are already D.E. duplicated at cell level.

**Comment by fairliereese** [comment]
Just commenting that this is how the new BioRad scATAC protocol works and it would be really great to see support for this in the future!
